glyphs-reader/src/propagate_anchors.rs

```diff
@@ -49,26 +49,41 @@ fn propagate_all_anchors_impl(glyphs: &mut BTreeMap<SmolStr, Glyph>) {
     let mut all_anchors = HashMap::new();
     for name in todo {
         let glyph = glyphs.get(&name).unwrap();
-        for layer in &glyph.layers {
-            let anchors = anchors_traversing_components(
-                glyph,
-                layer,
-                glyphs,
-                &all_anchors,
-                &mut num_base_glyphs,
-            );
+        for layer in glyph.layers.iter().chain(glyph.bracket_layers.iter()) {
+            let anchors =
+                anchors_traversing_components(glyph, layer, &all_anchors, &mut num_base_glyphs);
             maybe_log_new_anchors(&anchors, glyph, layer);
-            all_anchors.entry(name.clone()).or_default().push(anchors);
+            // if this is a bracket layer we use the actual axis values as the key,
+            // since it's possible that layers with the same axis values do not share
+            // the same layer id.
```

Comment:
This comment would go great on a key type!

Reply:
agree this feels a bit hacky, but a key type is also a challange: the issue is that we want the map to store both these custom keys (for bracket layers) as well as just master ids (uuid-type strings) and  then when we start mixing types like that it turns into a PITA because in the ID case you can't just borrow the name anymore, i.e. instead of going,

```rust
map.get(&layer.name)
```

you need to do,

```rust
map.get(&MyKey::JustAnId(layer.name.clone()))
```

and personally it doesn't quite feel worth it for a fairly self-contained internal API?